implementation/src/test/java/io/smallrye/config/ExpressionConfigSourceInterceptorTest.java

```diff
@@ -0,0 +1,79 @@
+package io.smallrye.config;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.NoSuchElementException;
+
+import org.eclipse.microprofile.config.Config;
+import org.junit.Test;
+
+public class ExpressionConfigSourceInterceptorTest {
+    @Test
+    public void simpleExpression() {
+        SmallRyeConfig config = (SmallRyeConfig) buildConfig("my.prop", "1234", "expression", "${my.prop}");
+
+        final String value = config.getValue("expression", String.class);
+        assertEquals("1234", value);
+    }
+
+    @Test
+    public void multipleExpressions() {
+        SmallRyeConfig config = (SmallRyeConfig) buildConfig("my.prop", "1234", "expression", "${my.prop}${my.prop}");
+
+        final String value = config.getValue("expression", String.class);
+        assertEquals("12341234", value);
+    }
+
+    @Test
+    public void composedExpressions() {
+        SmallRyeConfig config = (SmallRyeConfig) buildConfig("my.prop", "1234", "expression", "${${compose}}", "compose",
+                "my.prop");
+
+        final String value = config.getValue("expression", String.class);
+        assertEquals("1234", value);
+    }
+
+    @Test
+    public void defaultExpression() {
+        SmallRyeConfig config = (SmallRyeConfig) buildConfig("expression", "${my.prop:1234}");
+
+        final String value = config.getValue("expression", String.class);
+        assertEquals("1234", value);
+    }
+
+    @Test
+    public void defaultComposedExpression() {
+        SmallRyeConfig config = (SmallRyeConfig) buildConfig("expression", "${my.prop:${compose}}", "compose", "1234");
+
+        final String value = config.getValue("expression", String.class);
+        assertEquals("1234", value);
+    }
```

Comment:
Last thing (I think!): we need test cases for these cases:

* Property expression including `${foo:}` where `foo` does not exist (should expand to empty string)
* Property expression including `${foo}` where `foo` does not exist (should result in `NoSuchElementException`, preferably one which gives the path to the original key as well (e.g. "Could not expand property `foo` in property `bar` in property `baz`" or something of the sort).

Reply:
Sure!

Reply:
Interesting,

In the case of `${foo:}` we get a `Property foo not found`. This is because our default built in converter for Empty String values returns null.

Reply:
That's fine, in this case I'd expect it.  For the test you could do e.g. `123${foo:}456` and ensure that the result is `123456`.

Reply:
The expander of course has to treat `null` as empty though...

Reply:
To clarify, the expansion does work and expands to empty string.

Reply:
Ok, new push. Lets see now :)